### DISCUSSION TITLE
Bike-shedding: `guards` instead of `guard`

### DISCUSSION THREAD

**Discussion by devanshj** [statelyai/xstate#2420] [Ideas] [answered]
Afaik, v5 renames `cond` to `guard` but imo it should be `guards` to make it consistent with `actions` (Both have same shape and semantics except for guard requires the return type to be a `boolean`)

I was writing types for the implementations in txstate and realized this inconsistency. Instead of something like <code>\`${Type}s\`</code> I have to write `Type extends "action" ? "actions" : "guard"`.

Or alternatively it would be better to rename `actions` to `action` (along with `cond` to `guard`). Because `actions: "foo"` is much more awkward looking (as there is only one action but it says "actions") than `action: ["foo", "bar"]` (this too is weird that it has two actions but it says "action" though it could be seen as `["foo", "bar"]` represents one merged action). And `entry` and `exit` are also singular.

Again none of this is a huge deal, just was thinking if there's going to be a breaking change let's leverage it and get things right :P

**Comment by davidkpiano** [comment, marked as answer]
But `guard` is only one guard, and should only be one guard. 

You can't do this:
```diff
on: {
  EVENT: {
    target: 'somewhere',
-   guards: [guardOne, guardTwo]
+   guard: and([guardOne, guardTwo])
  }
}
```

**Reply by devanshj** [reply]
Oh shoot ahaha TIL. Makes sense. (Brb fixing txstate xP)
